### ISSUE TITLE
Incorrect JSC_WRONG_ARGUMENT_COUNT warning on String.prototype.toLocaleLowerCase

### ISSUE THREAD

**Issue by Fry-kun** [google/closure-compiler#2737]
Trying to compile 
``` es6
alert('Hello World'.toLocaleLowerCase('tr'));
```

at https://closure-compiler.appspot.com with advanced optimization level results in a warning:

```
JSC_WRONG_ARGUMENT_COUNT: Function String.prototype.toLocaleLowerCase: called with 1 argument(s). Function requires at least 0 argument(s) and no more than 0 argument(s). at line 1 character 6
alert('Hello World'.toLocaleLowerCase('tr'));
      ^
```

Per MDN, [toLocaleLowercase](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLocaleLowerCase) takes an optional argument that specifies the locale to use.

Same issue with .toLocaleUpperCase (mentioning in case anyone is searching for it)

**Comment by brad4d** [comment]
The "Browser compatibility" section of the MDN page indicates that only Firefox is known to allow the locale parameter. We won't want to update the closure-compiler externs files until this parameter is standardized.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLocaleUpperCase

**Comment by Fry-kun** [comment]
Actually Chrome supports this parameter, at least current version (v62) -- not sure how far back.
Tested in older browser (Firefox 52) -- it simply defaults to ignoring the argument, which is the "best" way of falling back in this case.
I think it's a good candidate for making an exception... OTOH, it's just a warning -- maybe rephrase it to say not all browsers support this argument?

**Comment by brad4d** [comment]
The trouble with updating our externs files is that it would basically be saying that we expect all browsers currently in use will be OK with the extra parameter and will not throw exceptions because of it.
It would encourage people to write code using this parameter, believing it to be perfectly safe, then possibly blowing up with an exception on some browser they didn't test.

Possible workaround:
```javascript
/**
 * @param {string} str
 * @param {string=} locale
 * @return {string}
 */
function toLocaleUpperCase(str, locale) {
  try {
    return str.toLocaleUpperCase(locale);
  } catch (e) {
    return str.toLocaleUpperCase();
  }
}
```

**Comment by ChadKillingsworth** [comment]
@brad4d This throws an exception? I thought the extra param was just silently ignored.

**Comment by Fry-kun** [comment]
I only tested one browser, an old version of Firefox -- and it silently ignores the param.
Some other browser (Opera? IE? etc?) may throw an error since this is a built-in method.
